spec/controllers/students_controller_spec.rb

```diff
@@ -604,6 +604,11 @@ def make_request(options = { student_id: nil, format: :pdf, from_date: '08/15/20
           expect(assigns(:event_notes)).to include(note)
           expect(assigns(:event_notes)).not_to include(restricted_note)
         end
+
+        it 'assigns the student\'s assesments correctly' do
```

Comment:
This is a good start! There's a bunch of logic that goes into the assignment of the `@student_assessments` variable:

```
    @student_assessments = student_assessments_by_date.each_with_object({}) do |student_assessment, hash|
      test = student_assessment.assessment
      test_name = case test.family
        when "STAR" then "#{test.family} #{test.subject} Percentile"
        else "#{test.family} #{test.subject}"
      end

      hash[test_name] ||= []

      result = case test.family
        when "MCAS" then student_assessment.scale_score
        when "Next Gen MCAS" then student_assessment.scale_score
        when "STAR" then student_assessment.percentile_rank
        when "DIBELS" then student_assessment.performance_level
        else student_assessment.scale_score
      end

      hash[test_name].push([student_assessment.date_taken,result])
    end.sort.to_h
```

Would be good to have test code that exercises more of this logic.

Reply:
@alexsoble What your thoughts on this version of the test, is there more I could be testing?

Another thing I wanted to mention is line `613`, I had to make the call again to be able to see these assessments for the student, and the test above (starting on line `601`) didn't need to call it again so that is puzzling to me.

Reply:
@gabivoicu I think you've made a good addition, the test covers more surface area now! 

I agree it's puzzling, you shouldn't have to call `make_request` again. One hypothesis is that some of the database relationships between the assessments and student assessments you've created may need a `reload` call to work properly. Take a look at this code sample from https://github.com/rspec/rspec-rails under "Model specs":

```
RSpec.describe Post, :type => :model do
  context "with 2 or more comments" do
    it "orders them in reverse chronologically" do
      post = Post.create!
      comment1 = post.comments.create!(:body => "first comment")
      comment2 = post.comments.create!(:body => "second comment")
      expect(post.reload.comments).to eq([comment2, comment1])
    end
  end
end
```

They use reload to make sure that the `post` object has access to its new `comments`.

That said, I don't think you should spend more than an hour or so on it — broader test coverage is the goal and if some of that coverage has quirkiness in it I think that's alright.